api/cmd/api/setup.go

```diff
@@ -83,8 +84,13 @@ func initMLPAPIClient(ctx context.Context, cfg config.MlpAPIConfig) mlp.APIClien
 	return mlp.NewAPIClient(mlpHTTPClient, cfg.APIHost, cfg.EncryptionKey)
 }
 
-func initFeastCoreClient(feastCoreURL string) core.CoreServiceClient {
-	cc, err := grpc.Dial(feastCoreURL, grpc.WithInsecure())
+func initFeastCoreClient(feastCoreURL, feastAuthAudience string) core.CoreServiceClient {
+	cred, err := feast.NewGoogleCredential(feastAuthAudience)
+	if err != nil {
+		panic(err)
+	}
+
+	cc, err := grpc.Dial(feastCoreURL, grpc.WithInsecure(), grpc.WithPerRPCCredentials(cred))
```

Comment:
You have to choose either `grpc.WithInsecure()` or `grpc.WithPerRPCCredentials(cred)` isn't it?
And we should add enabled/disabled flag

Reply:
need to add `gcpc.WIthInsecure()` because the feast credentials is not transport credentials.

Reply:
Got it